exporter/otelarrowexporter/factory.go

```diff
@@ -0,0 +1,143 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package otelarrowexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/otelarrowexporter"
+
+import (
+	"context"
+	"runtime"
+	"time"
+
+	arrowpb "github.com/open-telemetry/otel-arrow/api/experimental/arrow/v1"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configcompression"
+	"go.opentelemetry.io/collector/config/configgrpc"
+	"go.opentelemetry.io/collector/config/configopaque"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/exporter"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+	"google.golang.org/grpc"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/otelarrowexporter/internal/arrow"
+)
+
+const (
+	// The value of "type" key in configuration.
+	typeStr = "otelarrow"
```

Comment:
use the metadata.Type instead

Reply:
This is an example where you're reviewing code from the underlying OTLP exporter, why I've suggested an abbreviated review process (or a bypass). See https://github.com/open-telemetry/opentelemetry-collector/blob/bcb00a7b27fa66c7e0069b8d679f0581b4e9dd56/exporter/otlpexporter/factory.go#L20.

Reply:
yes, but that receiver doesn't have a metadata.yaml and metadata code generated. Please look at every other receiver in this repository, which we have painstakingly moved to use metadata.Type.

Reply:
Got it. Thanks.